driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala

```diff
@@ -56,25 +47,52 @@ object Driver extends DriverConfig {
 
   def apply() = new Driver()
 
-  def apply(flattenTables: Boolean) = new Driver(flattenTables: Boolean)
-
   lazy val defaultTimeout = Timeout(config .getDuration(DriverRetryDuration, MILLISECONDS), MILLISECONDS)
   lazy val defaultRetries = config.getInt(DriverRetryTimes)
+
+  def generateDefaultAuth = new Authentication("crossdata", "stratio")
+
 }
 
-class Driver(properties: java.util.Map[String, ConfigValue], flattenTables: Boolean) {
+/*
+ * ======================================== NOTE ========================================
+ * Take into account that every time the interface of this class is modified or expanded,
+ * the JavaDriver.scala has to be updated according to those changes.
+ * =======================================================================================
+ */
+
+class Driver(properties: java.util.Map[String, ConfigValue] = Map.empty[String, ConfigValue],
+             auth: Authentication = Driver.generateDefaultAuth,
+             flattenTables: Boolean = false) {
 
-  def this(properties: java.util.Map[String, ConfigValue]) = this(properties, false)
+  def this(serverHosts: java.util.List[String], auth: Authentication, flattenTables: Boolean) = this(
+    Map(DriverConfigHosts -> ConfigValueFactory.fromAnyRef(serverHosts)),
+    auth,
+    flattenTables)
 
-  def this(serverHosts: java.util.List[String]) =
-    this(Map(DriverConfigHosts -> ConfigValueFactory.fromAnyRef(serverHosts)), false)
+  def this(serverHosts: java.util.List[String], auth: Authentication) = this(
+    Map(DriverConfigHosts -> ConfigValueFactory.fromAnyRef(serverHosts)),
+    auth,
+    false)
 
-  def this(flattenTables: Boolean) = this(Map.empty[String, ConfigValue], flattenTables)
+  def this(serverHosts: java.util.List[String], flattenTables: Boolean) = this(
+    Map(DriverConfigHosts -> ConfigValueFactory.fromAnyRef(serverHosts)),
+    Driver.generateDefaultAuth,
+    flattenTables)
 
-  def this() = this(false)
+  def this(serverHosts: java.util.List[String]) = this(
+    Map(DriverConfigHosts -> ConfigValueFactory.fromAnyRef(serverHosts)),
+    Driver.generateDefaultAuth,
+    false)
 
   import Driver._
 
+  val driverSession = SessionManager.createSession(auth)
+
+  def securitizeCommand(command: SQLCommand): SecureSQLCommand = {
```

Comment:
If you move this conversion to `Driver` companion object and change its signature to implicit `implicit def securitizeCommand(command: SQLCommand): SecureSQLCommand`, you can pass `SQLCommand` where `SecureSQLCommand` is required and the user won't even notice it